### DISCUSSION TITLE
tropo_gacos: can not convert NaN to integer

### DISCUSSION THREAD

**Discussion by NMoghaddam** [insarlab/MintPy#1128] [General]
**Description of the problem**

In order to use GACOS for tropospheric correction, I downloaded the *.ztd files for my AOI; I managed to run the MintPy on the S1 time series with GACOS *.zdt files; however, it seems there is something wrong with the data values. I haven't requested GACOS files for every single scene but selected three for each year (a total of nine *.zdt files in Tiff format). The warnings are just for the missing GACOS files but the error came up after reading the first *.ztd file as such:

**Full script that generated the error**

```
correct_tropo = "Perform" in tropo_choice.value or "Rerun" in tropo_choice.value
era5_path = mint_path/"ERA5"
timeseries_era5_path = mint_path/"timeseries_ERA5.h5"
inputs_era5_path = mint_path/"inputs/ERA5.h5"

if "Delete" in tropo_choice.value:
    for f in [timeseries_era5_path, inputs_era5_path]:
        try:
            f.unlink()
        except FileNotFoundError:
            pass
    try:
        shutil.rmtree(era5_path)
    except FileNotFoundError:
        pass

if correct_tropo:
    set_troposhperic_correction_mintpy(config_path, "gacos")
    !smallbaselineApp.py $config_path --work-dir {mint_path} --dostep load_data
    !smallbaselineApp.py $config_path --work-dir {mint_path} --dostep correct_troposphere
else:
    set_troposhperic_correction_mintpy(config_path, "no")
    !smallbaselineApp.py $config_path --work-dir {mint_path} --dostep load_data
```

**Full error message**

```
Potential bug with data values in GACOS tiff files [as attached]
```

**System information**

* Operating system: EarthScope InSAR OpenSARLab 
* Python environment: Python 3.11 within the insar_analysis environment 
* MintPy version: MintPy version 1.5.2
* Your custom / default template file (if the bug is related to a specific dataset): 

![image](https://github.com/insarlab/MintPy/assets/13143710/8d5f6de0-ce13-4268-8045-b62fbea3b343)
![image (1)](https://github.com/insarlab/MintPy/assets/13143710/94294568-3dac-4b10-9527-bc37ec59cc57)

**Comment by welcome** [comment]
👋  Thanks for opening your first issue here! Please filled out the template with as much details as possible. We appreciate that you took the time to contribute!
Make sure you read our [contributing guidelines](https://github.com/insarlab/MintPy/blob/master/docs/CONTRIBUTING.md).

**Comment by yunjunz** [comment]
I am not sure why this error is happening. Could you debug at [tropo_gacos.py#L43](https://github.com/insarlab/MintPy/blob/db83a402e2d31b88c49ea53d61d7e68ebe34e9b2/src/mintpy/tropo_gacos.py#L43), or plot/print the `delay`, `length`, `width` variables?

**Comment by NMoghaddam** [comment]
I added the `print` command for `delay`,`length` and `width` in the [tropo_gacos.py#L43](https://github.com/insarlab/MintPy/blob/db83a402e2d31b88c49ea53d61d7e68ebe34e9b2/src/mintpy/tropo_gacos.py#L43) and run the workflow for GACOS... it seems that I can not get the value for each of those parameters, but a different error to the previous reported one as below. 

<img width="488" alt="ChangetoCode" src="https://github.com/insarlab/MintPy/assets/43155679/b746278d-dc4f-415f-a362-4d3a00b1795d">


<img width="761" alt="ERRORMessageA" src="https://github.com/insarlab/MintPy/assets/43155679/f2c9efa0-03d7-42df-82a7-25f93924137c">

I am not sure whether it has anything to do with the `calculate_delay_timeseries` and the following part:

<img width="478" alt="error" src="https://github.com/insarlab/MintPy/assets/43155679/5910a69d-1f76-4ca8-b20b-7123277ee4ef">

and if so what it does mean?!

**Comment by yunjunz** [comment]
The new issue should be due to the empty hdf5 file generated in the previous run. Removing it and re-run should get rid of this latter issue.

Converting this issue into discussion since I still could not re-product the reported error.
